### ISSUE TITLE
MatlabLexer cannot handle literal "function*" occurences

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments#1195]
(Original issue 1491 created by None on 2019-01-10T08:11:35.870765+00:00)

If the Matlab code contains a variable named ``function.*`` word anywhere which is not an actual function header, then it will be misinterpreted for example:

``` matlab
classdef ClassExample
    % function in the comment
    % _function_ in the comment
    % function_ in the comment

    methods
        function functionInTheName(obj)
            variable_function = 10;
            functionVariable = 10;
        end
    end
end
```

This causes a problem in the ``sphinxcontrib-matlab`` python package

**Comment by Anteru** [comment]
(Original comment by GulyasGergelyR on 2019-01-10T08:34:57.532351+00:00)

(Original poster)
Found the problem, The keyword token does not checks for the following character:
```
(r'^\s*function', Keyword, 'deffunc'),
```
which can be only a space or a square bracket so:
```
(r'^\s*function(?=[\s[])', Keyword, 'deffunc'),
```
seems to work.

**Comment by birkenfeld** [comment]
This was fixed by 06028bddcd2466b009363302669ee33b810ee185.
